### PR TITLE
Add persistent flag to Metric.add_state

### PR DESCRIPTION
**Pull request by teddykoker**
## What does this PR do?

Adds `persistent` flag to Metrics, to allow the user to disable state being
added to the `state_dict`.

### CODE REVIEW

**Comment by teddykoker** [comment]
Ugh, the `persistent` flag for `register_buffer` was not available in PyTorch 1.3

**Comment by SkafteNicki** [comment]
Can we add test for this? One metric class with and one without the persistent flag on their state, and then check their state dict.

**Comment by teddykoker** [comment]
> Can we add test for this? One metric class with and one without the persistent flag on their state, and then check their state dict.

Done :)

**Review by Borda — APPROVED** [review]
lgtm

**Review by ananyahjha93 — APPROVED** [review]

**Review by ydcjeff — APPROVED** [review]

**Comment by Vichoko** [comment]
Why would someone want to save a metric to the state_dict?
When you add metrics to the state_dict by default, you cannot load checkpoints any longer after you make changes to any of the metrics.
I feel pretty natural to add or update metrics of an already trained module. 

For example, in my issue #4666
